BlueTracker.SDK.Performance/Core/ApiWrapper.cs

```diff
@@ -19,21 +19,27 @@ public abstract class ApiWrapper
     {
         private readonly string _serverAddress;
         private readonly string _authorization;
+        private static HttpClient _httpClient;
+
+        /// <summary>
+        /// Singleton instance for every ApiClient to prevent depletion of connection pool
+        /// </summary>
+        private static HttpClient HttpClient => _httpClient ?? (_httpClient = new HttpClient());
 
         private const string DefaultServerAddress = "https://api.bluetracker.one";
 
         protected ApiWrapper(string serverAddress = null, string authorization = null)
         {
             _serverAddress = string.IsNullOrEmpty(serverAddress) ? GetServerAddress() : serverAddress;
             _authorization = string.IsNullOrEmpty(authorization) ? GetApiKey() : authorization;
+            HttpClient.BaseAddress = new Uri(_serverAddress);
```

Comment:
Hmmm... This could result in a dangerous race condition. The client is now static and shared between all instances of ApiWrapper. Imagine two instances are created at the same time with two different addresses. The addresses will be overwritten by each other in the client. The actual call of one instance might go into a different server than expected. We know that within an application run, the address is constant, but nevertheless. The static client should be completely stateless, i.e. it should not carry the base address. The base address should be set on the request level, so that each request contain the requires state.

1. Have a static HttpClient without the base address being set.
2. Store the base address in a private member variable for each ApiWrapper instance, which is set in the CTOR.
3. On each usage of the client, combine the base address with the requested route.
4. Set the combined URL for the request.
5. Execute the request.